lib/ddg-atbgen.js

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright (C) 2016 DuckDuckGo, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+"use strict";
+
+var ss = require("sdk/simple-storage");
+
+var atbTime = {
+    oneWeek     : 604800000,
+    oneDay      : 86400000,
+    oneHour     : 3600000,
+    oneMinute   : 60000,
+    estEpoch    : 1456290000000
+}
+
+var timeSinceEpoch = getTimeSinceEpoch(),
+    majorVersion = Math.ceil(timeSinceEpoch / atbTime.oneWeek),
```

Comment:
Should you recalculate these at the time of uninstall? `majorVersion` and `minorVersion` will be calculated when the extension loads and there could be cases that the browser has been running for more than a day. In that case it would report the uninstall as happening in the previous day.